modules/core/src/main/java/org/apache/ignite/internal/binary/BinaryObjectExImpl.java

```diff
@@ -245,7 +253,7 @@ private String toString(BinaryReaderHandles ctx, IdentityHashMap<BinaryObject, I
             IgniteThread.onForbidBinaryMetadataRequestSectionLeft();
         }
 
-        if (meta == null || !S.includeSensitive())
+        if (meta == null || sensitiveDataLogging != PLAIN)
             return S.toString(S.includeSensitive() ? BinaryObject.class.getSimpleName() : "BinaryObject",
```

Comment:
Should we hide class simpleName according to sensitiveDataLogging instead of S.includeSensitive?